prost-build/src/code_generator.rs

```diff
@@ -293,26 +293,29 @@ impl<'a> CodeGenerator<'a> {
         ));
 
         let prost_path = self.config.prost_path.as_deref().unwrap_or("::prost");
-        let string_path = format!("{}::alloc::string::String", prost_path);
-        let format_path = format!("{}::alloc::format", prost_path);
+        let string_path = format!("{prost_path}::alloc::string::String");
 
-        self.buf.push_str(&format!(
-            r#"fn full_name() -> {string_path} {{
-                {format_path}!("{}{}{}{}{{}}", Self::NAME)
-            }}"#,
+        let full_name = [
             self.package.trim_matches('.'),
             if self.package.is_empty() { "" } else { "." },
-            self.type_path.join("."),
+            &self.type_path.join("."),
             if self.type_path.is_empty() { "" } else { "." },
+            message_name,
+        ]
+        .concat();
```

Comment:
I think using `format!` similar to the original code is more readable. The formatting can be executed at compile time, of course.